### DISCUSSION TITLE
Can swagger UI be disabled?

### DISCUSSION THREAD

**Discussion by svghadi** [argoproj/argo-cd#16791] [Q&A]
Api docs are served on `<argocd-url>/swagger-ui` endpoint without authentication. Even though these docs are the same across different Argo CD instances and don't pose a security risk when exposed, some of the users are interested in knowing if we could disable this by using some flag? I tried finding it in docs, but no luck :frowning_face:

**Comment by svghadi** [comment]
Enhancement request: https://github.com/argoproj/argo-cd/issues/19780

**Comment by pasha-codefresh** [comment]
Unfortunately no, but should be pretty easy to support it 

https://github.com/argoproj/argo-cd/blob/a0a5a186d970063919e7b4d6e942e60cf43f6de7/server/server.go#L1076

**Comment by svghadi** [comment]
See https://github.com/argoproj/argo-cd/issues/19780#issuecomment-2467216319
